cli/global/middlewares/teardown.go

```diff
@@ -0,0 +1,18 @@
+package global_middlewares
+
+import (
+	"github.com/kubeshop/tracetest/cli/analytics"
+	global_setup "github.com/kubeshop/tracetest/cli/global/setup"
+	"github.com/spf13/cobra"
+)
+
+func WithTeardownMiddleware(setup *global_setup.Setup) Middleware {
+	return func(runFn RunFn) RunFn {
+		return func(cmd *cobra.Command, args []string) (string, error) {
```

Comment:
This command feels sometimes shady for me because it obliges us to know that we need to do a teardown. What happens if do we forget to call it and use `setup.Logger` or `analytics`? 

To think later: is there a way to keep these commands register to suffer the cleanup only if used? With that we could keep the golang behavior: 
```go
    resource := someAction()
    defer resourceCleanup()
```